test/common/export_results.c

```diff
@@ -18,17 +18,50 @@
 /** Maximum length of output path/name */
 #define MAX_FILENAME_LEN 192
 
+/** Maximum length of args string */
+#define MAX_ARGS_LEN 128
+
 typedef struct {
 	test_common_options_t common_options;
 
 	char filename[MAX_FILENAME_LEN];
 
 	FILE *file;
 
+	char args[MAX_ARGS_LEN];
+
 } test_export_gbl_t;
 
 static test_export_gbl_t gbl_data;
 
+static void test_common_extract_options(int argc, char *argv[])
+{
+	int i;
+	size_t ret = 0;
+	size_t current_len = strlen(gbl_data.args);
+	size_t remaining_space = MAX_ARGS_LEN - current_len;
+
+	if (argc > 1)
+		ret = snprintf(gbl_data.args + current_len, remaining_space, "%s", argv[1]);
+
+	if (ret >= remaining_space)
+		goto exit;
+
+	for (i = 2; i < argc; i++) {
+		current_len = strlen(gbl_data.args);
+		remaining_space = MAX_ARGS_LEN - current_len;
+		ret = snprintf(gbl_data.args + current_len, remaining_space, " %s", argv[i]);
+		if (ret >= remaining_space)
+			goto exit;
+	}
```

Comment:
The function could perhaps be simplified a bit. For example something like this:
```
	int ret;
	int len = 0;
	int remaining_space = sizeof(gbl_data.args);

	for (int i = 1; i < argc && remaining_space > 0; i++) {
		ret = snprintf(&gbl_data.args[len], remaining_space, " %s", argv[i]);
		if (ret >= remaining_space || ret < 0)
			goto exit;
		len += ret;
		remaining_space -= ret;
	}
...
```